pkg/cmd/issue/view/view.go

```diff
@@ -65,6 +69,8 @@ func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Comman
 	}
 
 	cmd.Flags().BoolVarP(&opts.WebMode, "web", "w", false, "Open an issue in the browser")
+	cmd.Flags().IntVarP(&opts.Comments, "comments", "c", 1, "View issue comments sorted by most recent")
+	cmd.Flags().Lookup("comments").NoOptDefVal = "30"
```

Comment:
I went with `NoOptDefVal` here which allows the `--comments` flag to be invoked with either `--comments` or `--comments=<number of comments>`. I thought this was a nice way to not have the user think about the number of comments they want to retrieve and just retrieve a sensible default number of comments.

The one downside is that `NoOptDefVal` makes it so that invoking with `--comments <number of comments>` is an error. This behavior differs from other integer flags, but I felt that it was worth the tradeoff.

Reply:
> The one downside is that `NoOptDefVal` makes it so that invoking with `--comments <number-of-comments>` is an error.

Yeah, the behavior of NoOptDefVal flags (basically, boolean-or-value flags) can be surprising and we usually try to avoid introducing them if possible. A good question to ask ourselves would be, could we get away with just using plain vanilla flags?

One thing that I think our users might notice is that the comment count is capped at 30 unless you explicitly pass a larger number as `--comments=999`. But, we display this in the UI: `Use --comments to view the full conversation`. That sentence could lead someone to believe that they will be able to see the full conversation, whereas they will only be shown the last part of it (if it's a giant conversation, like some headed threads are).

Reply:
I think we could use just plain vanilla flags. What do you think about switching to default retrieving the 3 (or 5) most recent comments and then removing the `NoOptDefVal`? I think this would satisfy the main use case of using `--comments` without specifying a specific number. 

I agree that the wording is a bit confusing. Two options I think we could go with are 
`Use --comments <total issue comments> to view the full conversation` or `Use --comments to view more of the conversation`.

Reply:
I'm a fan of `Use --comments to view more of the conversation`!

Reply:
I chatted with @ampinsk via Slack and we came up with a simplified approach since we agreed that the `NoOptDefValue` results in confusing behaviors and that having to specify the number of comments to retrieve each time is burdensome to the user. For this initial version the `--comments` flag will be a boolean flag that will be used to retrieve ALL the comments for an issue and won't be configurable. We will also be able to keep the `Use --comments to view the full conversation` language since it will be accurate.